# 1. State the Hypotheses:
Null Hypothesis (H₀): There is no significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and customer satisfaction levels. The two variables are independent.

Alternative Hypothesis (H₁): There is a significant association between the type of smart home device purchased and customer satisfaction levels. The two variables are dependent.

In [1]:
import numpy as np
from scipy.stats import chi2_contingency, chi2

# Step 1: Define the observed data (contingency table)
observed = np.array([
    [50, 70],   # Very Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
])

# Step 2: Perform the Chi-Square test
chi2_stat, p_value, dof, expected = chi2_contingency(observed)

# Step 3: Calculate the critical value
alpha = 0.05  # Significance level
critical_value = chi2.ppf(1 - alpha, dof)  # Critical value from Chi-Square distribution

# Step 4: Make a decision
if chi2_stat > critical_value:
    decision = "Reject the null hypothesis: There is a significant association."
else:
    decision = "Fail to reject the null hypothesis: No significant association."

# Step 5: Print all results
print("Observed Frequencies:")
print(observed)
print("\nExpected Frequencies:")
print(expected)
print(f"\nChi-Square Statistic: {chi2_stat}")
print(f"P-value: {p_value}")
print(f"Degrees of Freedom: {dof}")
print(f"Critical Value: {critical_value}")
print(f"\nDecision: {decision}")

Observed Frequencies:
[[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]

Expected Frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]

Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
Degrees of Freedom: 4
Critical Value: 9.487729036781154

Decision: Fail to reject the null hypothesis: No significant association.


# Conclusion:

The Chi-Square statistic (5.639) is less than the critical value (9.488), and the p-value (0.228) is greater than the significance level (α=0.05). Therefore, we fail to reject the null hypothesis, meaning there is no significant association between the type of smart home device and customer satisfaction levels.